processor/abci.go

```diff
@@ -746,6 +747,12 @@ func (app *App) OnCheckTx(ctx context.Context, _ tmtypes.RequestCheckTx, tx abci
 		app.log.Debug("entering checkTx", logging.String("tid", tx.GetPoWTID()), logging.String("command", tx.Command().String()))
 	}
 
+	if !app.exec.MarketsExpectKey(tx.PubKeyHex()) {
```

Comment:
I think we only want to do this check if the incoming transaction is a `SubmitOracleDataCommand`?

Reply:
I have corrected this here https://github.com/vegaprotocol/vega/pull/5534/commits/8ed6a685a32b5cdaf527f4bd083f72c17ff2a55a#diff-43cb75f5e3d4bfc8b8fc4540853a32c32346ec1fb151bbaed720d7bf07a4a139R822-R825
Thanks!